src/main/java/org/terasology/simpleFarming/systems/VineAuthoritySystem.java

```diff
@@ -118,14 +115,32 @@ public void onVinePlanted(OnSeedPlanted event, EntityRef definitionEntity, VineD
      */
     @ReceiveEvent
     public void onVineGrowth(DelayedActionTriggeredEvent event, EntityRef root, VineNodeComponent nodeComponent, VineDefinitionComponent vineComponent) {
-        if (nodeComponent.height != -1) {
-            if (nodeComponent.height < 20) {
+        doGrowVine(root, nodeComponent, vineComponent);
+    }
+
+    private void doGrowVine(EntityRef root, VineNodeComponent nodeComponent, VineDefinitionComponent vineComponent) {
+        if (nodeComponent.length != -1) {
+            if (nodeComponent.length < 20) {
                 recurseGrow(root, vineComponent);
             }
             resetDelay(root, vineComponent.minGrowTime, vineComponent.maxGrowTime);
         }
     }
 
+    @ReceiveEvent
+    public void onCheatGrowth(ActivateEvent event, EntityRef item, CheatGrowthComponent cheatGrowthComponent, ItemComponent itemComponent) {
+        if (!event.getTarget().hasComponent(VineDefinitionComponent.class) || !event.getTarget().hasComponent(VineNodeComponent.class)) {
```

Comment:
Could you please replace `event.getTarget` with a variable to make it clearer

Reply:
Deal.